tests/vpp.cpp

```diff
@@ -125,27 +135,53 @@ class VppTest
         return bool(m_allocator);
     }
 
+    void flushGPU()
+    {
+        std::deque<SharedPtr<VideoFrame> > frames;
+        SharedPtr<VideoFrame> f;
+        do {
+            f = m_allocator->alloc();
+            if (!f)
+                break;
+            vaSyncSurface(*m_display, f->surface);
+            frames.push_back(f);
+        } while (true);
+    }
+
     bool run()
     {
 
         SharedPtr<VideoFrame> src, dest;
         YamiStatus  status;
-        int count = 0;
+
+        FpsCalc fps;
+        uint32_t count = 0;
         while (m_input->read(src)) {
             dest = m_allocator->alloc();
             status = m_vpp->process(src, dest);
             if (status != YAMI_SUCCESS) {
                 ERROR("vpp process failed, status = %d", status);
                 return true;
             }
-            m_output->output(dest);
+
+            if (!m_skipDump)
+                m_output->output(dest);
             count++;
+            fps.addFrame();
+
+            if (count >= m_count)
+                break;
         }
         //flush output
         dest.reset();
         m_output->output(dest);
 
-        printf("%d frame processed\n", count);
+        //this will force all gpu operation finish. to get real performance
+        if (m_skipDump) {
+            flushGPU();
```

Comment:
flushGPU does not return any value, it's a wait function, wait all gpu command flushed. The only thing will be wrong in flushGPU will be vaSyncSurface, if vaSyncSurface return false, it means system trap to a serious issue, we can do nothing about his